### DISCUSSION TITLE
Adafruit ESP32 Huzzah Board compatibility

### DISCUSSION THREAD

**Discussion by alexholsgrove** [dawidchyrzynski/arduino-home-assistant#93] [Q&A] [answered]
Hi, I am trying to use this ESP32 board from Adafruit with Arduino Home Assistant, but have run into a few problems compiling. I have the library working well on an Arduino Nano 33 IoT board (some subtle differences such as using the WiFiNINA library), but cannot seem to get it to work on this board.

The main lines from my sketch:

`#include <ArduinoHA.h>
#include <SPI.h>
#include <WiFi.h>
#include <Wire.h>

WiFiClient client;

HADevice device;
HAMqtt mqtt(client, device);
HASensor tempProbe1("probe1");
HASensor tempProbe2("probe2");
HASensor batteryLevel("batteryLevel");
HABinarySensor batteryLow("batteryLow", "", false);`

And the errors when verifying:

`In file included from /Users/Alex/Development/Arduino/libraries/home-assistant-integration/src/ArduinoHA.h:5,
                 from /Users/Alex/Development/ESP32/ESP32_BBQ_Monitor/ESP32_BBQ_Monitor.ino:9:
/Users/Alex/Development/Arduino/libraries/home-assistant-integration/src/HAMqtt.h:25:12: error: 'arduino' has not been declared
     HAMqtt(arduino::Client& netClient, HADevice& device);
            ^~~~~~~
/Users/Alex/Development/Arduino/libraries/home-assistant-integration/src/HAMqtt.h:25:27: error: expected ')' before '&' token
     HAMqtt(arduino::Client& netClient, HADevice& device);
           ~               ^
                           )
/Users/Alex/Development/Arduino/libraries/home-assistant-integration/src/HAMqtt.h:227:5: error: 'arduino' does not name a type; did you mean 'Arduino_h'?
     arduino::Client& _netClient;
     ^~~~~~~
     Arduino_h
/Users/Alex/Development/ESP32/ESP32_BBQ_Monitor/ESP32_BBQ_Monitor.ino:36:27: error: no matching function for call to 'HAMqtt::HAMqtt(WiFiClient&, HADevice&)'
 HAMqtt mqtt(client, device);
                           ^
In file included from /Users/Alex/Development/Arduino/libraries/home-assistant-integration/src/ArduinoHA.h:5,
                 from /Users/Alex/Development/ESP32/ESP32_BBQ_Monitor/ESP32_BBQ_Monitor.ino:9:
/Users/Alex/Development/Arduino/libraries/home-assistant-integration/src/HAMqtt.h:15:7: note: candidate: 'constexpr HAMqtt::HAMqtt(const HAMqtt&)'
 class HAMqtt
       ^~~~~~
/Users/Alex/Development/Arduino/libraries/home-assistant-integration/src/HAMqtt.h:15:7: note:   candidate expects 1 argument, 2 provided
/Users/Alex/Development/Arduino/libraries/home-assistant-integration/src/HAMqtt.h:15:7: note: candidate: 'constexpr HAMqtt::HAMqtt(HAMqtt&&)'
/Users/Alex/Development/Arduino/libraries/home-assistant-integration/src/HAMqtt.h:15:7: note:   candidate expects 1 argument, 2 provided
Multiple libraries were found for "WiFi.h"
  Used: /Users/Alex/Library/Arduino15/packages/esp32/hardware/esp32/2.0.4/libraries/WiFi
  Not used: /Users/Alex/Development/Arduino/libraries/WiFiNINA
exit status 1

Compilation error: no matching function for call to 'HAMqtt::HAMqtt(WiFiClient&, HADevice&)'`

Can you please advise what I am doing wrong, or if I can provide any more information. Thank you

**Comment by dawidchyrzynski** [comment, marked as answer]
Hi @alexholsgrove,

It looks like you have some custom modifications in the HAMqtt header. Try to revert all changes.

**Reply by alexholsgrove** [reply]
Hi David, my apologies - I had modified the HAMQTT.h as I had a similar problem (initially) to the user here: https://github.com/dawidchyrzynski/arduino-home-assistant/issues/82 and had followed [bogdan2000's](https://github.com/bogdan2000) fix. This resolved the problem when I was using an Arduino Nano 33 IoT board, but then failed when compiling for my ESP32. I've reverted the changes and it works find on the ESP32 board. I'm yet to re-compile for the Nano but will reply back if that's still an issue.

Thank you for your work to provide this library for the community - very much appreciated.

**Reply by dawidchyrzynski** [reply]
You can apply the below fix to get the build working on both platforms. Just add this piece of code in the HAMqtt.h file, right before the `class HAMqtt {`

```cpp
#ifdef __SAMD21G18A__
using namespace arduino;
#endif
```

I'm still working on the 2.0.0 version of the lib where this fix will be already included.
